Prozess.txt

```diff
@@ -0,0 +1,240 @@
+		Das ergebnis eines Programms wird dabei als Eingabe eines anderen Programms verwendet.
+		In der shell wird dies durch den "|" realisiert. So kann man mehrere befehle hintereinander Ausführen.
+
+/dev/null:
+	Verhält sich wie eine leere Datei, mit dem Unterschied, dass eingehende Datenströme verworfen werden. 
+	Beim Lesezugriff wir ein einzelnes End-of-file Zeichen ausgegebn.
+	Verwendung:
+		-Verwerfung von nicht interessanten Datenströmen, z.B. asugaben vom Prozesor
+		
+Administratorrechte Prozess:
+	sudo chmod n+s (programmpfad) bewirkt dass das programm mit den rechten der Benutzergruppe des SuperUsers
+	ausgeführt wird. Generell steht sudo für SuperUser do, und führt den nachfolgenden befehl 
+	mit Administratorrechten aus(falls vorhanden).
+	
+.EXE Datei und ihre Struktur:
+	Bei einer .exe datei handelt es sich unter Windows-Basierten Systemen um eine Ausführbare datei.
+	Sie besteht aus:
+		-Ausführbarem Code, welcher in der .exe als .text vorliegt
+		-Datensektionen, liegt als .data vor
+		-Ressourcensektionen (.rsrc)
+		-Importdatensektion (.idata)
+		-Debuginfotmationssektion(.debug)
+
+Interprozesskommunikation:
+	Erfolgt über verschiedene Verfahren:o clarify
+		-Shared Memory:
+			-Mehrere Prozesse greifen auf einen gemeinsamen Datenspeicher zu, bsp. auf
+			 gemeinsame Bereiche des Arbeitsspeichers.
+		-Message Queue:
+			-Nachrichten von einem Prozess an eine Nachrichtenanlage geschickt, von
+			 wo ein anderer Prozess sie abholen kann.
+
+	Wichtig:
+		Vermeidung von Race Conditions:
+			Kein Zugriff von mehreren Prozessen gleichzeitig auf einen Speicher.
+			
+Posix:
+	Ist ein IEEE STandard zur einheitlichen Bereitstellung von Schnittstellen in Unixbasierten Systemen.
+	Dadurch ist es möglich dass ein Programm welches auf dem POSIX Standard basiert, höchstwahrscheinlich
+	auf einer vielzahl von Betriebssystemen laufen kann. Verwendet man Betriebssystem Spezifische API's,
+	wird man auf erhebliche Probleme stoßen, falls man die Anwendung auf ein anderes Betriebssystem portieren will.
+	
+	
+POSIX-Shell:
+	Die BASH ist nicht standardisiert. Sie wird durch die in ihr implementierten Funktionen
+	definiert. Die POSIX-Shell hingegen wird durch ihren Standard bestimmt, welcher es ermöglicht die POSIX-Shell
+	auf allen Unix Systemen zum Laufen zu bringen.
+	
+Shells:
+	Bourne (UNIX), POSIX, BASH, csh, zsh, cmd.exe, PowerShell etc
+	
+Globbing:
+	Globbing beschreibt das Anwenden von Wildcard-Patterns(platzhaltern), um so auf dateien zuzugreifen ohne ihren genauen dateinahmen kennen zu müssen.
+	Dabei haben die verscheidenen PLatzhalter folgende bedeutung:
+		das Fragezeichen (?) ist ein platzhalter für genau ein Zeichen.
+		das Sternchen (*) ist ein platzhalter für beliebig viele Zeichen.
+		die Raute (#) ist ein platzhalter für einen numerischen Wert.
+		
+Umgebungsvariablen:
+	Sind konfigurierbare Variablen die Pfade zu Programmen oder Daten. Dadurch ist es möglich ein programm in der shell direkt
```

Comment:
Clarification needed:

1. In which program are those variables defined?
2. How does a process gain access to environment variables?
3. Why are those variables called environment variables?
4. Why are environment variables useful? Can you name a few use cases?